include/deal.II/lac/linear_operator.h

```diff
@@ -1095,15 +1238,198 @@ linear_operator(const OperatorExemplar &operator_exemplar, const Matrix &matrix)
+                               MatrixInterfaceWithoutVmultAdd<Range, Domain, Payload>>::type().
                                operator()(return_op, matrix);
 
   return return_op;
 }
 
+
 //@}
 
+
+#ifdef DEAL_II_WITH_TRILINOS
+
+// Forward declarations:
+
+namespace TrilinosWrappers
+{
+  class SparseMatrix;
+  class PreconditionBase;
+
+  namespace internal
+  {
+    namespace LinearOperator
+    {
+      class TrilinosPayload;
+      class TrilinosPreconditionPayload;
+    }
+  }
+
+  /**
+   * @name Creation of a LinearOperator
+   */
+//@{
+
+
+  /**
+   * @relates LinearOperator
+   *
+   * Return a LinearOperator that is the identity of the vector space @p Range.
+   *
+   * This function is the equivalent of the dealii::identity_operator, but
+   * ensures full compatibility with Trilinos operations by preselecting the
+   * appropriate template parameters.
+   *
+   * @author Jean-Paul Pelteret, 2016
+   *
+   * @ingroup TrilinosWrappers
+   */
+  template <typename Range>
+  inline LinearOperator<Range, Range, TrilinosWrappers::internal::LinearOperator::TrilinosPayload>
+  identity_operator(const std::function<void(Range &, bool)> &reinit_vector)
+  {
+    typedef TrilinosWrappers::internal::LinearOperator::TrilinosPayload Payload;
+    return dealii::identity_operator<Range, Payload>(reinit_vector);
+  }
+
+
+  /**
+   * @relates LinearOperator
+   *
+   * Return a nulled variant of the LinearOperator @p op.
+   *
+   * This function is the equivalent of the dealii::null_operator, but
+   * ensures full compatibility with Trilinos operations by preselecting the
+   * appropriate template parameters.
+   *
+   * @author Jean-Paul Pelteret, 2016
+   *
+   * @ingroup TrilinosWrappers
+   */
+  template <typename Range,
+            typename Domain = Range>
+  inline LinearOperator<Range, Domain, TrilinosWrappers::internal::LinearOperator::TrilinosPayload>
+  null_operator(const LinearOperator<Range, Domain, TrilinosWrappers::internal::LinearOperator::TrilinosPayload> &op)
+  {
+    typedef TrilinosWrappers::internal::LinearOperator::TrilinosPayload Payload;
+    return dealii::null_operator<Range, Domain, Payload> (op);
+  }
+
+
+  /**
+   * @relates LinearOperator
+   *
+   * A function that encapsulates generic @p matrix objects, based on an
+   * @p operator_exemplar, that act on a compatible Vector type into a
+   * LinearOperator.
+   *
+   * This function is the equivalent of the dealii::linear_operator, but
+   * ensures full compatibility with Trilinos operations by preselecting the
+   * appropriate template parameters.
+   *
+   * @author Jean-Paul Pelteret, 2016
+   *
+   * @ingroup TrilinosWrappers
+   */
+  template <typename Range, typename Domain = Range,
+            typename Matrix>
+  inline LinearOperator<Range, Domain, TrilinosWrappers::internal::LinearOperator::TrilinosPayload>
+  linear_operator(const TrilinosWrappers::SparseMatrix &operator_exemplar, const Matrix &matrix)
```

Comment:
Overloading the `linear_operator` constructors is perfectly fine (as long as it can be done with forward declarations).

Regarding your question about ensuring type compatibility - we can add a couple of `static_assert`s in the `LinearOperator` class ensuring that we select the correct payloads. I suggest we do that later after we got a bit of experience with the changes.